include/seqan3/range/view/translation.hpp

```diff
@@ -45,292 +45,684 @@
- * \tparam irng_t The type of the range being processed.
- * \param irange The range being processed.
- * \returns A range containing aminoacid sequence. See below for the properties of the returned range.
+
+/*!\name Alphabet related views
+ * \{
+ */
+
+/*!\brief A view that translates nucleotide into aminoacid alphabet for one of the six frames.
+ * \tparam urng_t The type of the range being processed.
+ * \param[in] urange The range being processed.
+ * \param[in] tf A value of seqan3::tanslation_frames that indicates the desired frames.
+ * \returns A range containing frames with aminoacid sequence. See below for the properties of the returned range.
  * \ingroup view
  *
  * \details
  *
- * This view can be used to translate nucleotide sequences into single forward aminoacid sequences.
-
- * \par View properties
+ * This view can be used to translate nucleotide sequences into aminoacid sequences (see translation_frames for possible combination of frames).
  *
- * |                     | `irng_t` (range input type)           | `rrng_t` (range return type)                              |
- * |---------------------|---------------------------------------|-----------------------------------------------------------|
- * | range               | seqan3::input_range_concept           | seqan3::view_concept + all range concepts met by `irng_t` |
- * | `range_reference_t` | seqan3::nucleotide_concept            |                                                           |
+ * ### View properties
  *
- * * `irng_t` is the type of the range modified by this view (input).
+ * | range concepts and reference_t      | `urng_t` (underlying range type)      | `rrng_t` (returned range type)                     |
+ * |-------------------------------------|:-------------------------------------:|:--------------------------------------------------:|
+ * | seqan3::input_range_concept         | *required*                            | *preserved*                                        |
+ * | seqan3::forward_range_concept       |                                       | *preserved*                                        |
+ * | seqan3::bidirectional_range_concept |                                       | *preserved*                                        |
+ * | seqan3::random_access_range_concept | *required*                            | *preserved*                                        |
+ * |                                     |                                       |                                                    |
+ * | seqan3::view_concept                |                                       | *guaranteed*                                       |
+ * | seqan3::sized_range_concept         | *required*                            | *preserved*                                        |
+ * | seqan3::bounded_range_concept       |                                       | *preserved*                                        |
```

Comment:
The return type is definitely bounded, but I am not 100% sure whether it is also required, I think not (that would then make it *guaranteed*).

You can try by taking a vector and applying `| ranges::view::take_exactly(n)` on it, where `n` is the size of the range. This will produces something that is random_acces, sized, but not bounded. Then stick this into a `view::translate_single` and see if it works. (It should). Then you know the bounded property is *guaranteed*, otherwise it is preserved and you need to make bounded also a requirement for `urng_t`.